unittest/bindings/factory.py

```diff
@@ -747,9 +747,16 @@ def calc(self, data, x, u=None):
         data.cost = data.activation.a_value
 
     def calcDiff(self, data, x, u=None):
-        data.residual.Rx[:] = self.state.Jdiff(
-            self.xref, x, crocoddyl.Jcomponent.second
-        )[0]
+        # The old code was looking like this.
+        # But, the std::vector<Eigen::MatrixXd> returned by Jdiff is destroyed
+        # before the assignment.
+        # To avoid this issue, we store the std::vector in a variable.
+        # data.residual.Rx[:] = self.state.Jdiff(
+        #     self.xref, x, crocoddyl.Jcomponent.second
+        # )[0]
+
+        diff = self.state.Jdiff(self.xref, x, crocoddyl.Jcomponent.second)
+        data.residual.Rx[:] = diff[0]
```

Comment:
Could you avoid breaking our Python API?

I am afraid this will create a lot of stress in many projects based on Crocoddyl.

This seems to be related to changes in eigenpy as you can see here: https://github.com/loco-3d/crocoddyl/blob/devel/bindings/python/crocoddyl/utils/vector-converter.hpp#L12

Reply:
Hello, I don't think the API is broken here (I just store the Jdiff return in a temporary variable).

But this fix hide something really weird. We should investigate if some change in pinocchio 3 std::vector binding can impact the crocoddyl std::vector management.

Reply:
Thanks for investigating this. But if I remember correctly, Crocoddyl is using eigenpy std::vector bindings.

Reply:
I guess this is non blocking then.

Reply:
ref #1255